### DISCUSSION TITLE
Can you use `.route()` without chaining?

### DISCUSSION THREAD

**Discussion by alshdavid** [tokio-rs/axum#2760] [Q&A]
### Summary

Not a major issue, just curious. A personal preference of mine is to avoid method chaining but the `Router` implementation seems to prevent non-chained usage.

I am wondering if there is a way list routes like this:
```rust
#[tokio::main]
async fn main() {
  let mut app = Router::new();

  app.route("/", get(root));
  app.route("/users", post(create_user));

  let listener = tokio::net::TcpListener::bind("0.0.0.0:8080").await.unwrap();
  axum::serve(listener, app).await.unwrap();
}
```

### axum version

0.7.5

**Comment by mladedav** [comment]
There isn't, `route` takes `self` by value.

I don't think there is a technical limitation that would prevent using `&mut self` for that, but for example `with_state` produces a `Router` with a different generic parameter so that has to be passed by value.

**Comment by alshdavid** [comment]
Thanks!
